packages/syntax-core/src/TapArea/TapArea.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { type ReactNode, forwardRef } from "react";
+import classNames from "classnames";
+import styles from "./TapArea.module.css";
+
+type TapAreaProps = {
+  /**
+   * The children to be rendered inside the tap area.
+   */
+  children?: ReactNode;
+  /**
+   * The label to be used for accessibility
+   */
+  accessibilityLabel?: string;
+  /**
+   * Test id for the tap area
+   */
+  "data-testid"?: string;
+  /**
+   * If `true`, the tap area will be disabled
+   *
+   * @defaultValue false
+   */
+  disabled?: boolean;
+  /**
+   * If `true`, the tap area will be full width
+   */
+  fullWidth?: boolean;
```

Comment:
If TapArea doesn't have a size prop, is it worth considering to always make this full width? I haven't really thought this entirely through, but I imagine we could control this similar to how `Divider` is, which is the full size of the parent

Reply:
@somethiiing in the case of `TapArea`, we probably don't want it to be always full width, otherwise, you would always need to set a container around it (probably a `Box`) when using it inline.

Example from the docs where we want to show them side-by-side:

![image](https://github.com/Cambly/syntax/assets/127199/ea4204ab-742a-41de-8039-071c457a64f2)
![image](https://github.com/Cambly/syntax/assets/127199/68624122-b028-45f6-84b5-5268a9d7fca5)